R/cmapplot.R

```diff
@@ -17,73 +17,125 @@
 #' @keywords internal
 "_PACKAGE"
 
-cmapplot_globals <- new.env(parent = emptyenv())  # An environment for storing any global variables
-
-
-# Default font handling ---------------------------------------------------
-
-# sizes
-cmapplot_globals$fsize <- list(
-  big = 17,
-  reg = 14,
-  sml = 11
-)
-
-if (.Platform$OS.type == "windows") {
-
-  # Check for Whitney
-  all_fonts <- sysfonts::font_files()
-  whitney_fonts <- dplyr::filter(all_fonts,
-    family %in% c("Whitney Medium", "Whitney Book", "Whitney Semibold") & face == "Regular"
+#' cmapplot global variables
+#'
+#' Expose a list of predefined variables for use by the cmapplot package and its
+#' users.
+#'
+#' @export
+cmapplot_globals <- list(
+
+  ## Colors
+  colors = list(
+    blackish = "#222222"
+  ),
+
+  ## Font sizes
+  fsize = list(
+    big = 17,
+    reg = 14,
+    sml = 11
+  ),
+
+  ## Base typefaces -- modified later by .onLoad()
+  font = list(
+    strong = list(family="Arial", face="bold"),
+    regular = list(family="Arial", face="plain"),
+    light = list(family="Arial", face="plain")
```

Comment:
I'm still at 50% comprehension with fonts in R, but it's interesting to me that this works considering that WindowFonts() doesn't display a font called "Arial", but rather:

```
$sans
[1] "TT Arial"
```

Is it just different on Macs?

Reply:
I'm not sure if it "works" or if it's just silently failing and and then reverting to R's default font... Arial!